dot-studio/golang

```diff
@@ -0,0 +1,160 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+document "setup_go_workspace" <<DOC
+  Setup the Go Workspace. (requires \$pkg_scaffolding to be set)
+
+  This function will leverage the scaffolding-go configured at \$pkg_scaffolding.
+  It will install it and use the go_before callback to setup the Go Workspace.
+  If there are no libraries/callbacks, it will still configure it by linking the /src
+  directory into the Go Workspace.
+DOC
+function setup_go_workspace() {
+  # Verify that we are using the Go Scaffolding
+  if [[ ! "$pkg_scaffolding" =~ "scaffolding-go" ]]; then
+    echo "ERROR: Unable to setup Go Workspace."
+    echo "${FUNCNAME[0]}: requires the 'pkg_scaffolding' variable to exist in your plan.sh"
+    return 2;
+  fi;
+
+  # Verify if the Go Workspace has been already configured
+  [[ $scaffolding_go_workspace_src  ]] && return 0;
+
+  # Currently the scaffolding-go has all the logic to setup the
+  # Go Workspace, lets leverage all its functionality
+  echo "=> Configuring Go Workspace. [please wait]"
+  export scaffolding_go_gopath=${scaffolding_go_gopath:-/hab/cache/src/go};
+  install $pkg_scaffolding >/dev/null;
+  source "$(hab pkg path $pkg_scaffolding)/lib/"*
+  export PATH="$PATH:${scaffolding_go_gopath}/bin"
+
+  # Verify if we have the go_before callback
+  # If not lets setup the workspace ourself
+  go_before_callback=scaffolding_go_before
+  if [ -n "$(type -t $go_before_callback)" ] && [ "$(type -t $go_before_callback)" = function ]
+  then
+    eval "$go_before_callback";
+  else
+    mkdir -p "${scaffolding_go_workspace_src}/${scaffolding_go_base_path}";
+    if [[ ! -e "$scaffolding_go_pkg_path" ]]; then
+      ln -s /src "$scaffolding_go_pkg_path";
+    fi;
+  fi;
+
+  echo "=> Available scaffolding variables:";
+  echo "\$scaffolding_go_gopath        - $scaffolding_go_gopath";
+  echo "\$scaffolding_go_workspace_src - $scaffolding_go_workspace_src";
+  echo "\$scaffolding_go_pkg_path      - $scaffolding_go_pkg_path";
+  return 0;
+}
+
+document "install_go_tool" <<DOC
+  Install the specified go tool(s).
+
+  @(arg:*) The array of packages you wish to install.
+
+  Example 1 :: Install the listed go tool generating a static linked binary
+  ---------------------------------------------------------------------------------
+  GO_STATIC_BIN=true install_go_tool github.com/golang/protobuf/protoc-gen-go
+
+  Example 2 :: Install a set of go tools
+  --------------------------------------------------------------------------
+  go_tools=(
+    github.com/golang/lint/golint
+    github.com/golang/dep/cmd/dep
+    github.com/derekparker/delve/cmd/dlv
+  )
+  install_go_tool \${go_tools[@]}
+DOC
+function install_go_tool() {
+  setup_go_workspace
+  local go_cmd="go get -u"
+
+  if [[ $GO_STATIC_BIN == true ]]; then
+    install core/musl >/dev/null
```

Comment:
we pipe this to `>/dev/null` enough that I feel like we should just put that as the default...thoughts?

Reply:
Let me see the output.

Reply:
I keep seeing it to many times:
```
=> Installing Go Tool 'foo'
Installing core/musl
=> Installing Go Tool 'bar'
Installing core/musl
=> Installing Go Tool 'lala'
Installing core/musl
```

I think this case makes sense to hide it since if you do:
```
install_go_tool foo
install_go_tool baar
install_go_tool lala
```
You will see the `Installing x` multiple times.